### PR TITLE
fix: Close button for Add pane when tabs are disabled

### LINKED ISSUE #33655: [Task]: Fix Side by Side flag features for airgapped
### SubTasks

- [ ] Add pane should have close button to close it without tabs
- [ ] Remove the Edit actions in the header from the feature flag rules

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

- Segment add header gets a close button if the side by side flag is not enabled
- Remove the edit menu item from header (not just for feature flag)

Fixes #33655

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9218769515>
> Commit: f92c783b754781f2e6801cf62ca55db92bf8019d
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9218769515&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/client/src/pages/Editor/IDE/EditorPane/JS/Add.tsx [hunk 1]
```diff
@@ -53,6 +57,7 @@ const AddJS = ({ containerProps, innerContainerProps }: AddProps) => {
         {...innerContainerProps}
       >
         <SegmentAddHeader
+          onCloseClick={closeAddJS}
```
Comment (rishabhrathod01) [inline app/client/src/pages/Editor/IDE/EditorPane/JS/Add.tsx:60]: We only need to close the button for the Query Add Pane correct?

File: app/client/src/pages/Editor/EditorName/NavigationMenuData.ts [hunk 2]
```diff
@@ -141,27 +136,6 @@ export const GetNavigationMenuData = ({
       type: MenuTypes.MENU_DIVIDER,
       isVisible: true,
     },
-    !isSideBySideFlagEnabled && {
```
Comment (rishabhrathod01) [inline app/client/src/pages/Editor/EditorName/NavigationMenuData.ts:144]: Could you help me understand why are we removing this and what would be it's impact?

Comment (hetunandu) [reply app/client/src/pages/Editor/EditorName/NavigationMenuData.ts:144]: We removed this to fix the Information architecture of the header. Undo and Redo actions do not exist on the app name level and should not be in that menu. We have already fixed tests around this when Side by Side went into open beta

**Review by rishabhrathod01 — COMMENTED** [review]
Code changes LGTM, testing it in local

**Review by rishabhrathod01 — APPROVED** [review]

**Comment by NilanshBansal** [comment]
Re-triggering only GIT tests as only one `GitBugs_Spec.ts` is failing.
image

**Pull request merged**